### PR TITLE
Feature/mqtt base structure

### PR DESCRIPTION
**Pull request by saschadoemer**

### CODE REVIEW

File: agrirouter-api-dotnet-standard-impl/Service/Messaging/Vcu/OnboardVcuService.cs [hunk 1]
```diff
@@ -17,18 +17,15 @@ namespace Agrirouter.Impl.Service.Messaging.Vcu
...
-        private readonly EncodeMessageService _encodeMessageService;
+        private readonly IMessagingService<MessagingParameters> _messagingService;
 
         /// <summary>
         /// Constructor.
         /// </summary>
         /// <param name="messagingService">-</param>
-        /// <param name="encodeMessageService">-</param>
-        public OnboardVcuService(MessagingService messagingService, EncodeMessageService encodeMessageService)
+        public OnboardVcuService(HttpMessagingService messagingService)
```
Comment (Frank-Wiebeler) [inline agrirouter-api-dotnet-standard-impl/Service/Messaging/Vcu/OnboardVcuService.cs:26]: Why are you using HttpMessagingService here; in all other classes it's the general IMessagingService

File: agrirouter-api-dotnet-standard-impl/Service/Messaging/ListEndpointsUnfilteredService.cs [hunk 2]
```diff
@@ -13,9 +13,7 @@ public class ListEndpointsUnfilteredService : ListEndpointsBaseService
         /// Constructor.
         /// </summary>
         /// <param name="messagingService"></param>
-        /// <param name="encodeMessageService"></param>
-        public ListEndpointsUnfilteredService(MessagingService messagingService,
-            EncodeMessageService encodeMessageService) : base(messagingService, encodeMessageService)
+        public ListEndpointsUnfilteredService(HttpMessagingService messagingService) : base(messagingService)
```
Comment (Frank-Wiebeler) [inline agrirouter-api-dotnet-standard-impl/Service/Messaging/ListEndpointsUnfilteredService.cs:16]: Why are you using HttpMessagingService here; in all other classes it's the general IMessagingService

File: agrirouter-api-dotnet-standard-impl/Service/Common/MqttMessagingService.cs [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        {
+            _mqttClient = mqttClient;
+        }
+
+        /// <summary>
+        /// Send message to the AR using the given message parameters.
+        /// </summary>
+        /// <param name="messagingParameters">Messaging parameters.</param>
+        /// <returns>-</returns>
+        /// <exception cref="CouldNotSendHttpMessageException">Will be thrown if the message could not be send.</exception>
```
Comment (Frank-Wiebeler) [inline agrirouter-api-dotnet-standard-impl/Service/Common/MqttMessagingService.cs:37]: CouldnotSendMqttMessageException?

File: agrirouter-api-dotnet-standard-impl/Service/Messaging/Abstraction/SendMessageBaseService.cs [hunk 4]
```diff
@@ -38,7 +34,7 @@ public MessagingResult Send(SendMessageParameters sendMessageParameters)
 
             if (string.IsNullOrWhiteSpace(sendMessageParameters.Base64MessageContent))
             {
-                throw new CouldNotSendMessageException(HttpStatusCode.BadRequest,
+                throw new CouldNotSendHttpMessageException(HttpStatusCode.BadRequest,
```
Comment (Frank-Wiebeler) [inline agrirouter-api-dotnet-standard-impl/Service/Messaging/Abstraction/SendMessageBaseService.cs:37]: That's not really an Http-Error, could better be a neutral CouldNotSendMessageException that is not Protocol specific

File: agrirouter-api-dotnet-standard-impl/Service/Messaging/Abstraction/SendMessageBaseService.cs [hunk 5]
```diff
@@ -125,17 +121,18 @@ public IEnumerable<string> ChunkMessageContent(string base64MessageContent, int
             var chunks = new List<string>();
             var source = Encoding.UTF8.GetBytes(base64MessageContent);
             byte[] chunk;
-            for(var i = 0; i < source.Length; i+=chunkSize)
+            for (var i = 0; i < source.Length; i += chunkSize)
             {
                 if (i + chunkSize > source.Length) continue;
```
Comment (Frank-Wiebeler) [inline agrirouter-api-dotnet-standard-impl/Service/Messaging/Abstraction/SendMessageBaseService.cs:126]: I'ld prefer 

`if(i+ chunkSize > source.Length) {
      continue;
}`

or even 

`if( i + chunkSize < = source.Length){
                   chunk = new byte[chunkSize];
                Array.Copy(source, i, chunk, 0, chunkSize);
                chunks.Add(Encoding.UTF8.GetString(chunk)); 
}
`

Those one-liner-Statements without {} lead to issues in code maintenance quite often

Comment (Frank-Wiebeler) [reply agrirouter-api-dotnet-standard-impl/Service/Messaging/Abstraction/SendMessageBaseService.cs:126]: Should mark the new code, no idea why it's on the old code.

Comment (saschadoemer) [reply agrirouter-api-dotnet-standard-impl/Service/Messaging/Abstraction/SendMessageBaseService.cs:126]: This is a common pattern in .NET - therefore no fix applied.

**Pull request merged**